lstchain/reco/dl1_to_dl2.py

```diff
@@ -417,3 +429,85 @@ def apply_models(dl1, classifier, reg_energy, reg_disp_vector, custom_config={})
     dl2['gammaness'] = probs
     return dl2
 
+
+
+def get_source_dependent_parameters(data, config={}):
+
+    """Get parameters for source-dependent analysis .
+
+    Parameters:
+    -----------
+    data: Pandas DataFrame
+    config: dictionnary containing configuration
+    
+    """
+
+    src_dep_params = pd.DataFrame(index=data.index)
+
+    is_simu = 'mc_type' in data.columns
+    
+    if is_simu:
+        if (data['mc_type'] == 0).all():
+            data_type = 'mc_gamma'
+        else:
+            data_type = 'mc_proton'
+    else:
+        data_type = 'real_data'
+    
+    expected_src_pos_x_m, expected_src_pos_y_m = get_expected_source_pos(data, data_type, config)
+
+    src_dep_params['expected_src_x'] = expected_src_pos_x_m
+    src_dep_params['expected_src_y'] = expected_src_pos_y_m
+    
+    src_dep_params['dist'] = np.sqrt((data['x'] - expected_src_pos_x_m)**2 + (data['y'] - expected_src_pos_y_m)**2)
+    src_dep_params['time_gradient_from_source'] = data['time_gradient'] * np.sign(data['x'] - expected_src_pos_x_m)
+    src_dep_params['skewness_from_source'] = data['skewness'] * np.sign(data['x'] - expected_src_pos_x_m)
+    
+    disp, miss = camera_to_shower_coordinates(expected_src_pos_x_m, expected_src_pos_y_m, data['x'], data['y'], data['psi'])
+    alpha = np.rad2deg(np.arctan2(miss, disp))
+    alpha[alpha<0] = 180 + alpha[alpha<0]
+    alpha[alpha>90] = 180 - alpha[alpha>90]
+    src_dep_params['alpha'] = alpha
+
+    return src_dep_params
+
+
+def get_expected_source_pos(data, data_type, config):
+
+    """Get expected source position for source-dependent analysis .
+
+    Parameters:
+    -----------
+    data: Pandas DataFrame
+    data_type: 'mc_gamma','mc_proton','real_data'
+    config: dictionnary containing configuration
+    
+    """
+
+    #For gamma MC, expected source position is actual one for each event
+    if data_type == 'mc_gamma':
+        expected_src_pos_x_m = data['src_x'].values
+        expected_src_pos_y_m = data['src_y'].values
+
+    #For proton MC, nominal source position is one written in config file
+    if data_type == 'mc_proton':
+        
+        focal_length = OpticsDescription.from_name('LST').equivalent_focal_length
+        
+        expected_src_pos = utils.sky_to_camera((data['mc_alt_tel'].values + config['mc_nominal_source_x_deg']) * u.deg,
```

Comment:
Use `u.Quantity(data['mc_alt_tel'].values + config['mc_nominal_source_x_deg'], u.deg, copy=False)` to be more efficient here.

Also linebreak after the first ( and then just indend by 4 spaces fixes the line length issue, like this:

```
function(
    arg1,
    arg2,
)
```

Reply:
thanks, done!